#!/usr/bin/env python
PACKAGE = "bobi_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("left_Kp",    double_t,    0, "Left proportional gain", 0.3,  0.0, 100.0)
gen.add("right_Kp",    double_t,    0, "Right proportional gain", 0.3,  0.0, 100.0)
gen.add("left_Ki",    double_t,    0, "Left integral gain", 0.0,  0.0, 100.0)
gen.add("right_Ki",    double_t,    0, "Right integral gain", 0.0,  0.0, 100.0)
gen.add("left_Kd",    double_t,    0, "Left derivative gain", 0.02,  0.0, 100.0)
gen.add("right_Kd",    double_t,    0, "Right derivative gain", 0.02,  0.0, 100.0)

gen.add("left_ub",    double_t,    0, "Left motor upper bound (m/s)", 0.40,  0.0, 0.60)
gen.add("right_ub",    double_t,    0, "Right motor upper bound (m/s)", 0.40,  0.0, 0.60)
gen.add("left_lb",    double_t,    0, "Left motor lower bound (m/s)", -0.40,  -0.60, 0)
gen.add("right_lb",    double_t,    0, "Right motor lower bound (m/s)", -0.40,  -0.60, 0)

gen.add("scaler_left",    double_t,    0, "Scaling coefficient for the left motor PID", 1.0,  0.00001, 100)
gen.add("scaler_right",    double_t,    0, "Scaling coefficient for the right motor PID", 1.0,  0.00001, 100)


gen.add("verbose",   bool_t,   0, "Verbose",  True)

exit(gen.generate(PACKAGE, "velocity_controller_node", "VelocityControl"))